portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -3,31 +3,87 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Date;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
 
 public class Comment {
     public static List<Comment> COMMENTS = new ArrayList<>();
-    
-    private int id;
+
+    private Long id;
     private String text;
-    private Date date_posted;
-    private Integer parent_id;
+    private String datePosted;
+    private Long parentId;
 
     private Comment(String text){
-        this.id = COMMENTS.size();
         this.text = text;
-        this.date_posted = new Date();
+        this.datePosted = convertDateToString(new Date());
     }
 
-    private Comment(String text, int parent_id){
+    private Comment(String text, long parentId){
         this(text);
-        this.parent_id = parent_id;
+        this.parentId = parentId;
+    }
+
+    private Comment(long id, String text, String datePosted, Long parentId){
+        this.id = id;
+        this.text = text;
+        this.datePosted = datePosted;
+        this.parentId = parentId;
     }
 
     public static void AddComment(String text){
```

Comment:
java methods usually don't have the first letter capitalized (i.e. addComment instead of AddComment). ditto elsewhere.